libs/dtm/dtm_recovery.py

```diff
@@ -107,40 +118,56 @@ def perform_ops(self, workload_info: list, queue, skipread: bool = True,
-                                       obj_name_pref=workload['obj_name_pref'],
-                                       obj_size=workload['obj_size'],
-                                       skip_cleanup=skipcleanup,
-                                       skip_write=True,
-                                       skip_read=skipread,
-                                       validate=validate,
-                                       log_file_prefix=f"read_workload_{workload['obj_size']}mb",
-                                       end_point=S3_CFG["s3_url"],
-                                       validate_certs=S3_CFG["validate_certs"])
-                self.log.info("Workload: %s objects of %s with %s parallel clients ",
-                              workload['num_sample'], workload['obj_size'],
-                              workload['num_clients'])
-                self.log.info("Log Path %s", resp[1])
-                log_path = resp[1]
-                if s3bench.check_log_file_error(resp[1]):
-                    results.append(False)
-                    break
-                else:
-                    results.append(True)
+                if not skipread or validate:
+                    resp = s3bench.s3bench(self.access_key,
+                                           self.secret_key,
+                                           bucket=workload['bucket'],
+                                           num_clients=workload['num_clients'],
+                                           num_sample=workload['num_sample'],
+                                           obj_name_pref=workload['obj_name_pref'],
+                                           obj_size=workload['obj_size'],
+                                           skip_cleanup=True,
+                                           skip_write=True,
+                                           skip_read=skipread,
+                                           validate=validate,
+                                           log_file_prefix=f"read_workload_{workload['obj_size']}b",
+                                           end_point=S3_CFG["s3_url"],
+                                           validate_certs=S3_CFG["validate_certs"])
+                    self.log.info("Workload: %s objects of %s with %s parallel clients ",
+                                  workload['num_sample'], workload['obj_size'],
+                                  workload['num_clients'])
+                    self.log.info("Log Path %s", resp[1])
+                    if s3bench.check_log_file_error(resp[1]):
+                        self.log.error("Error found in log path: %s", resp[1])
+                        results.append(False)
+                        break
+                    else:
+                        results.append(True)
+                if not skipcleanup:
+                    # Delete all objects from the bucket
+                    resp = self.s3t_obj.object_list(bucket_name=workload['bucket'])
```

Comment:
is this restricted to only objects from one bucket or can we extend to bucket list?

Reply:
This is iterated for workload, hence all object from buckets send in workload will be deleted in case of skipcleanup = False.